memory/index.go

```diff
@@ -96,55 +96,64 @@ func (idx *Index) NewLookup(ctx *sql.Context, ranges ...sql.Range) (sql.IndexLoo
 			case sql.RangeType_Null:
 				rangeColumnExpr = expression.NewIsNull(idx.Exprs[i])
 			case sql.RangeType_GreaterThan:
-				lit, typ := getType(sql.GetRangeCutKey(rce.LowerBound))
-				rangeColumnExpr = expression.NewNullSafeGreaterThan(idx.Exprs[i], expression.NewLiteral(lit, typ))
+				if sql.RangeCutIsBinding(rce.LowerBound) {
+					lit, typ := getType(sql.GetRangeCutKey(rce.LowerBound))
+					rangeColumnExpr = expression.NewGreaterThan(idx.Exprs[i], expression.NewLiteral(lit, typ))
+				} else {
+					rangeColumnExpr = expression.NewNot(expression.NewIsNull(idx.Exprs[i]))
+				}
 			case sql.RangeType_GreaterOrEqual:
 				lit, typ := getType(sql.GetRangeCutKey(rce.LowerBound))
-				rangeColumnExpr = expression.NewNullSafeGreaterThanOrEqual(idx.Exprs[i], expression.NewLiteral(lit, typ))
+				rangeColumnExpr = expression.NewGreaterThanOrEqual(idx.Exprs[i], expression.NewLiteral(lit, typ))
 			case sql.RangeType_LessThan:
 				lit, typ := getType(sql.GetRangeCutKey(rce.UpperBound))
-				rangeColumnExpr = expression.NewNullSafeLessThan(idx.Exprs[i], expression.NewLiteral(lit, typ))
+				rangeColumnExpr = or(
+					expression.NewLessThan(idx.Exprs[i], expression.NewLiteral(lit, typ)),
+					expression.NewIsNull(idx.Exprs[i]),
```

Comment:
`select * from t where c0 < 3` does not return nulls, so I'm assuming it doesn't get translated to a `RangeType_LessThan`